metricflow/test/fixtures/manifest_fixtures.py

```diff
@@ -37,22 +37,64 @@
 logger = logging.getLogger(__name__)
 
 
-class SemanticManifestName(Enum):
-    """Names of the semantic manifests used in testing. Listed under test/fixtures/semantic_manifest_yamls."""
-
-    AMBIGUOUS_RESOLUTION_MANIFEST = "ambiguous_resolution_manifest"
-    # Not included as it has intentional errors for running validations.
-    # CONFIG_LINTER_MANIFEST = "config_linter_manifest"
-    CYCLIC_JOIN_MANIFEST = "cyclic_join_manifest"
-    DATA_WAREHOUSE_VALIDATION_MANIFEST = "data_warehouse_validation_manifest"
-    EXTENDED_DATE_MANIFEST = "extended_date_manifest"
-    JOIN_TYPES_MANIFEST = "join_types_manifest"
-    MULTI_HOP_JOIN_MANIFEST = "multi_hop_join_manifest"
-    PARTITIONED_MULTI_HOP_JOIN_MANIFEST = "partitioned_multi_hop_join_manifest"
-    NON_SM_MANIFEST = "non_sm_manifest"
-    SCD_MANIFEST = "scd_manifest"
-    SIMPLE_MANIFEST = "simple_manifest"
-    SIMPLE_MULTI_HOP_JOIN_MANIFEST = "simple_multi_hop_join_manifest"
+@dataclass(frozen=True)
+class SemanticManifestSetupPropertySet:
+    """Describes a semantic manifest used in testing."""
+
+    # Name corresponds to a directory under test/fixtures/semantic_manifest_yamls.
+    semantic_manifest_name: str
+    # Specify a separate start value for each semantic manifest to reduce snapshot thrash when one semantic manifest
+    # is modified. i.e. without this, modifying the first semantic manifest might cause all IDs in snapshots associated
+    # with semantic manifests following the first one to change.
+    id_number_space: IdNumberSpace
+
+
+class SemanticManifestSetup(Enum):
+    """Enumeration of semantic manifests that defined in YAML files and used for testing."""
+
+    AMBIGUOUS_RESOLUTION_MANIFEST = SemanticManifestSetupPropertySet(
+        semantic_manifest_name="ambiguous_resolution_manifest",
+        id_number_space=IdNumberSpace.for_block(0),
+    )
+    # Not including CONFIG_LINTER_MANIFEST as it has intentional errors for running validations.
+    CYCLIC_JOIN_MANIFEST = SemanticManifestSetupPropertySet(
+        semantic_manifest_name="cyclic_join_manifest", id_number_space=IdNumberSpace.for_block(1)
+    )
```

Comment:
Ok, this refines the consistency further, but I'm still not sure if this solves the consistency problem altogether.